src/overlay/OverlayManagerImpl.cpp

```diff
@@ -119,10 +119,17 @@ OverlayManagerImpl::connectTo(std::string const& peerStr)
 }
 
 void
-OverlayManagerImpl::connectToImpl(PeerRecord pr, bool evenIfConnected)
+OverlayManagerImpl::connectToImpl(PeerRecord pr, bool force)
 {
     mConnectionsAttempted.Mark();
-    if (evenIfConnected || !getConnectedPeer(pr.getAddress()))
+
```

Comment:
per my previous comment on this one: this doesn't seem to work right.

* for normal connection attempts you don't want to connect at all if there is any connection (inbound or outbound). we're already not selecting those peers anyways from `tick` as far as I can tell (more of a safeguard).
* in the context of promotions, you want to force the connection only if it's not already connected outbound for some reason.

This is why I said earlier that the `force` flag should be a `forceoutbound` flag (meaning if it's already connected inbound connect anyways).

I *think* that the current implementation is already filtering upstream, so those are just clarification in the code on what we expect to happen: if a peer is already connected outbound we don't want to reconnect to it (what force as implemented would cause).